es6/README.md

```diff
@@ -0,0 +1,406 @@
+
+return `We saw ${count} errors in the ${category} category'
+```
+
+## Do not nest ternary expressions
+
+Ternary expressions can be helpful, but can also lead to unclear code. If multiple ternary operators are used in a single expression, it is clearer to use explicit `if` and `else` blocks.
+
+```javascript
+// Bad
+let name = (item 
+	? item.name 
+	: (defaultItem 
+		? defaultItem.name
+		: defaultName
+	)
+)
+
+// Good
+let name
+if (item) {
+	name = item.name
+} else if (defaultItem) {
+	name = defaultItem.name
+} else {
+	name = defaultName
+}
+```
+
+
+# Arrow Functions
+
+## Always use parentheses in arrow function definitions
+
+This makes the syntax consistent between single-argument arrow functions and multiple-argument (and zero-argument!) arrow functions.
+
+```javascript
+// bad!
+a.map(item => { return item * 2; })
+
+// good!
+a.map((item) => { return item * 2; })
+```
+
+## Do not use a placeholder parameter in a zero-parameter arrow function
+
+```javascript
+// bad!
+_ => console.log('This happened')
+
+// good!
+() => console.log('This happened')
+```
+
+## Space out the arrow function operator
+```javascript
+// Bad
+(item)=>{ console.log(item) }
+
+// Good
+(item) => { console.log(item) }
+```
+
+## Prefer arrow functions for callbacks
+
+If the callback does not use its *own* `this` value (as opposed to one from the enclosing scope), it should be an arrow function.
+
+```javascript
+// Bad
+return fetch('http://mysite.com/api')
+    .then(function(response) {
+        return response.json()
+    })
+    
+// Good
+return fetch('http://mysite.com/api')
+    .then((response) => {
+        return response.json()
+    })
+
+// Also good
+return fetch('http://mysite.com/api')
+    .then((response) => response.json())
+    
+// Still fine
+element.addEventListener('click', function(event) {
+	console.log(this.id)
+})
+```
+
+# Variable Declarations
+
+## Prefer `let` and `const` to `var`
+
+The new block-scoped declarations in ES6 are more predictable than the function-scoped `var` declaration in ES5. We should use them instead.
+
+```javascript
+// Bad
+var stack = []
+var item
+var MAXIMUM_RETRY_COUNT = 5
+
+// Good
+let stack = []
+let item
+const MAXIMUM_RETRY_COUNT = 5
+```
+
+## Prefer `const` when possible
+
+Make sure to declare values that are not to be modified as `const`. This will help ensure they are not modified inadvertently.
+
+## Declare variables individually
+
+This is more readable and prevents unexpected results when commenting out lines.
+
+```javascript
+// Bad
+let x = 5,
+    y = 10,
+    z = 100
+    
+// Good
+let x = 5
+let y = 10
+let z = 100
+```
+
+## Prefer rest parameters to `arguments`
+
+```javascript
+// Bad
+// Notice that the `arguments` object has no `reduce` method
+const mySum = () => {
+	let sum = 0
+	for (let i = 0; i < arguments.length; i++) {
+		sum += arguments[i]
+	}
+	return sum
+}
+
+// Good
+// values is an Array so it has all the array methods
+const mySum = (...values) => {
```

Comment:
This example is a bit confusing to me since a `mySum` method would probably only take one argument, the array to sum, so it might not be the best example for a rest parameter